src/graph/CompactionExecutor.cpp

```diff
@@ -0,0 +1,83 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "http/HttpClient.h"
+#include "graph/CompactionExecutor.h"
+#include "process/ProcessUtils.h"
+#include "webservice/Common.h"
+
+#include <folly/executors/Async.h>
+#include <folly/futures/Future.h>
+
+namespace nebula {
+namespace graph {
+
+CompactionExecutor::CompactionExecutor(Sentence *sentence,
+                               ExecutionContext *ectx) : Executor(ectx) {
+    sentence_ = static_cast<CompactionSentence*>(sentence);
+}
+
+Status CompactionExecutor::prepare() {
+    return Status::OK();
+}
+
+void CompactionExecutor::execute() {
+    auto status = checkIfGraphSpaceChosen();
+    if (!status.ok()) {
+        DCHECK(onError_);
+        onError_(std::move(status));
+        return;
+    }
+    auto *mc = ectx()->getMetaClient();
+    auto  addresses = mc->getAddresses();
+    auto  metaHost = network::NetworkUtils::intToIPv4(addresses[0].first);
+    auto spaceName = ectx()->rctx()->session()->spaceName();
+    auto  spaceId = ectx()->rctx()->session()->space();
+
+    auto func = [metaHost, spaceName, spaceId]() {
```

Comment:
It seems you need to compact all storage service, whether compact the meta service is not so important. BTW, why not use the compact interface in NebulaStore instead of use http client?